File: tests/example-my-class.rb
# Define the class
class MyClass

    def initialize(x, y, z, t)
        @x = x
        @y = y
        @z = z
        @t = t
    end

    def transform(v)
        @x = @x + v
        @y = "hello " + @y
        @z.sort!
        @t = !@t
        result = {}
        self.instance_variables.each do |var|
            result[var.to_s.delete("@")] = self.instance_variable_get(var)
        end
        return result
    end

end

# Example object
$my_obj = MyClass.new(10, "world", [3, 1, 2], true)

File: tests/example-my-class.py
# Define the class
class MyClass:

    def __init__(self, x, y, z, t):
        self.x = x
        self.y = y
        self.z = z
        self.t = t

    def transform(self, v):
        self.x = self.x + v
        self.y = "hello " + self.y
        self.z = list(sorted(self.z))
        self.t = not self.t
        return self.__dict__

# Example object
my_obj = MyClass(10, "world", [3, 1, 2], True)
